mautic/mautic
# Process contact import through form to validate data

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | (could be viewed as so)
| Related user documentation PR URL | na
| Related developer documentation PR URL | na
| Issues addressed (#s or URLs) | #2261 #2273
| BC breaks? | n
| Deprecations? | n  

**Note that all new features should have a related user and/or developer documentation PR in their respective repositories.** 

### Required
#### Description:

This PR uses the same form when importing contacts that is used for the API endpoint and the UI. This ensures that data is validated and matches what's expected (such as select options, country, state, etc). 

#### Steps to test this PR:
1. Create a CSV with some contact entries. Add a state column - have some that are valid (Virginia) and some that are not (Mars). 
2. Import the CSV
3. The rows with the bad state should not import and display the error in the list of post import report.  The rows with good data should be imported.

### As applicable
#### Steps to reproduce the bug:
1. Repeat the above before the PR is applied. The contacts with the bad states will be imported with the bad values. Edit the contact and note that the state field will be empty (because it's value doesn't match anything in the list).
